### PR TITLE
PD-4648 Create initial terraform and pipeline

### PR DESCRIPTION
**Pull request by dehe1**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,29 +9,40 @@ setup: # Set up project
...
+		start
 
-restart: stop start # Restart project
+stop: project-stop
 
-log: project-log # Show project logs
+log: project-log
 
-test: # Test project
+unit-test: # Test project
```
Comment (terado) [inline Makefile:26]: We've fixed make run-unit-test now so this is not necessary

File: Makefile [hunk 2]
```diff
@@ -40,46 +51,17 @@ clean: # Clean up project
...
-run-integration-test:
-	echo TODO: $(@)
+		if [ $(BUILD_ID) == 0 ]; then
+			container=roaddistance-lambda
+		else
+			container=roaddistance-lambda-$(BUILD_ID)
+		fi
+		docker exec \
+			$$container \
+			python -m pytest -q tests/unit/$(TEST_FILE) -k "$(NAME)"
```
Comment (terado) [inline Makefile:64]: This does not work. See above, you've removed the working version. Might be worth a chat?

File: build/docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    environment:
+      DRD_ENDPOINT: "http://localhost"
+      DRD_BASICAUTH: ""
+      DRD_MOCK_MODE: "True"
+    ports:
+      - 9000:8080
+    volumes:
+      - $APPLICATION_DIR/tests:/var/task/tests
+      - $APPLICATION_DIR:/var/task/application
+
```
Comment (terado) [inline build/docker/docker-compose.yml:17]: We also need 
```suggestion
      - $APPLICATION_DIR/mock:/var/task/mock
```

File: Makefile (hunk 1 above)
Comment (dehe1) [reply Makefile:26]: Updated to reflect

File: Makefile (hunk 2 above)
Comment (dehe1) [reply Makefile]: Merged the two together, working now

File: build/docker/docker-compose.yml (hunk 3 above)
Comment (dehe1) [reply build/docker/docker-compose.yml:18]: Added in to the compose file

**Review by forjayssake — DISMISSED** [review]

File: Makefile [hunk 4]
```diff
@@ -40,46 +46,21 @@ clean: # Clean up project
...
-		docker exec roaddistance-lambda \
-			python -m pytest -rA -q tests/unit/$(TEST_FILE) -k "$(NAME)"
-	fi
-
-run-smoke-test:
-	echo TODO: $(@)
-
-run-integration-test:
-	echo TODO: $(@)
+		if [ $(BUILD_ID) == 0 ]; then
```
Comment (stefaniuk) [inline Makefile:52]: This seems to depend on a running container. Is there a need for that? Could we instead use `make docker-run IMAGE=...`?

**Review by stefaniuk — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (dehe1) [reply Makefile:52]: Thanks, will take a look

**Review by stefaniuk — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Review by forjayssake — APPROVED** [review]

**Review by terado — APPROVED** [review]

**Pull request merged**